### PR TITLE
feat: add DepositTradeHelper skeleton

### PR DESCRIPTION
**Pull request by ramenforbreakfast**

### CODE REVIEW

**Comment by ramenforbreakfast** [comment]
Tests for other contracts are failing due to something going wrong with the snapshot reset. Feel free to review the contents since its unrelated to the contract itself.

File: apps/smart-contracts/core/contracts/DepositTradeHelper.sol [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  ICollateral private immutable _collateral;
+  IERC20Permit private immutable _baseToken;
+  ISwapRouter private immutable _swapRouter;
+  uint256 public constant override POOL_FEE_TIER = 10000;
+
+  constructor(ICollateral collateral, ISwapRouter swapRouter) {
+    _collateral = collateral;
+    _baseToken = IERC20Permit(address(collateral.getBaseToken()));
+    _swapRouter = swapRouter;
+    IERC20(address(collateral.getBaseToken())).approve(
```
Comment (XavierEkkel) [inline apps/smart-contracts/core/contracts/DepositTradeHelper.sol:17]:
```suggestion
    collateral.getBaseToken().approve(
```
base token should already be in `IERC20` form

File: apps/smart-contracts/core/contracts/DepositTradeHelper.sol [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+contract DepositTradeHelper is IDepositTradeHelper, SafeOwnable {
+  ICollateral private immutable _collateral;
+  IERC20Permit private immutable _baseToken;
+  ISwapRouter private immutable _swapRouter;
+  uint256 public constant override POOL_FEE_TIER = 10000;
+
+  constructor(ICollateral collateral, ISwapRouter swapRouter) {
+    _collateral = collateral;
+    _baseToken = IERC20Permit(address(collateral.getBaseToken()));
```
Comment (XavierEkkel) [inline apps/smart-contracts/core/contracts/DepositTradeHelper.sol:15]:
```suggestion
    _baseToken = IERC20Permit(collateral.getBaseToken());
```
will work I think?

**Review by XavierEkkel — APPROVED** [review]
LGTM with the snapshot logic taken out (can re-introduce in a separate PR)

File: apps/smart-contracts/core/contracts/DepositTradeHelper.sol (hunk 2 above)
Comment (ramenforbreakfast) [reply apps/smart-contracts/core/contracts/DepositTradeHelper.sol:15]: No

**Pull request merged**